### PR TITLE
Add empty-to-one `decode` method

### PR DESCRIPTION
**Pull request by NicoMulet**
### TL;DR

Add empty-to-one `decode` method to `JSONAPIDecoder`.

### Context

As stated in [JSONAPI specifications](https://jsonapi.org/format/#document-top-level):
> Primary data MUST be either:
> - a single [resource object](https://jsonapi.org/format/#document-resource-objects), a single [resource identifier object](https://jsonapi.org/format/#document-resource-identifier-objects), or null, for requests that target single resources
> - an array of [resource objects](https://jsonapi.org/format/#document-resource-objects), an array of [resource identifier objects](https://jsonapi.org/format/#document-resource-identifier-objects), or an empty array ([]), for requests that target resource collections

The library is currently not able to decode an empty-to-one resource (represented by `{ "data": null }`), and the `JSONAPIDecoder.decode` is throwing a decoding error in such case.
This PR aims to add a new `decode` method which will be able to decode empty-to-one resources.

### Summary of Changes

- Add `decode<R>(_: R?.Type, from data: Data) throws -> R? where R: ResourceIdentifiable & Decodable`

### How to Test
N/A

### Demo
```swift
let json = #"{ "data": null }"#.data(using: .utf8)
let article = try JSONAPIDecoder().decode(Article?.self, from: json) // article == nil
```
This example was previously resulting in a decoding error.

### CODE REVIEW

File: Tests/JSONAPITests/ResourceTests.swift [hunk 1]
```diff
@@ -112,6 +112,17 @@ final class ResourceTests: XCTestCase {
...
+	func testDecodeNullSingle() throws {
+		// given
+		let json = try XCTUnwrap(#"{ "data": null }"#.data(using: .utf8))
+
+		// when
+		let article = try JSONAPIDecoder().decode(Article?.self, from: json)
+
+		// then
+		XCTAssertNil(article)
+	}
```
Comment (gonzalezreal) [inline Tests/JSONAPITests/ResourceTests.swift]: It could be nice to add a test case for the optional overload but with an actual resource instead of `null`.

**Review by gonzalezreal — APPROVED** [review]
Looks good!

Maybe we should add another test case for the overload but with a valid resource instead of `nil`.
